src/Service/Doctrine/ResultProvider.php

```diff
@@ -34,6 +34,16 @@ public function getResultForQuery(ConfiguredQuery $configuredQuery, Pager $pager
 
         $result = $this->buildResult($analysedQuery, $pager);
 
+        if ($configuredQuery->getItemTransformer() !== null) {
```

Comment:
Could we move this block into a private method?

Now we kinda have different layers of details in this main method - as a reader, I'd like to first understand what's done in more abstract naming, only then dig into the details how exactly that's done (and only when needed).

Reply:
Extracted to a private function, set aside if check.